apis/kueue/v1beta1/provisioningrequestconfig_types.go

```diff
@@ -57,6 +58,58 @@ type ProvisioningRequestConfigSpec struct {
 	// +listType=set
 	// +kubebuilder:validation:MaxItems=100
 	ManagedResources []corev1.ResourceName `json:"managedResources,omitempty"`
+
+	// retryStrategy defines strategy for retrying ProvisioningRequest.
+	// If null, then the default configuration is applied with the following parameter values:
+	// BackoffLimitCount:  3
+	// BackoffBaseSeconds: 60 - 1 min
+	// BackoffMaxSeconds:  1800 - 30 mins
+	//
+	// To switch off retry mechanism
+	// set retryStrategy.backoffLimitCount to 0.
+	//
+	// +optional
+	RetryStrategy *ProvisioningRequestRetryStrategy `json:"retryStrategy,omitempty"`
+}
+
+var (
+	DefaultRetryStrategy = ProvisioningRequestRetryStrategy{
+		BackoffLimitCount:  ptr.To(int32(3)),
+		BackoffBaseSeconds: 60,   // 1 min
+		BackoffMaxSeconds:  1800, // 30 min
+	}
+)
+
+type ProvisioningRequestRetryStrategy struct {
+	// BackoffLimitCount defines the maximum number of re-queuing retries.
+	// Once the number is reached, the workload is deactivated (`.spec.activate`=`false`).
+	// When it is null, the workloads will repeatedly and endless re-queueing.
+	//
+	// Every backoff duration is about "b*2^(n-1)+Rand" where:
+	// - "b" represents the base set by "BackoffBaseSeconds" parameter,
+	// - "n" represents the "workloadStatus.requeueState.count",
+	// - "Rand" represents the random jitter.
+	// During this time, the workload is taken as an inadmissible and
+	// other workloads will have a chance to be admitted.
+	// By default, the consecutive requeue delays are around: (60s, 120s, 240s, ...).
+	//
+	// +optional
+	BackoffLimitCount *int32 `json:"backoffLimitCount,omitempty"`
+
+	// BackoffBaseSeconds defines the base for the exponential backoff for
+	// re-queuing an evicted workload.
+	//
+	// Defaults to 60.
+	// +required
+	// +kubebuilder:default=60
+	BackoffBaseSeconds int32 `json:"backoffBaseSeconds"`
```

Comment:
There is a tiny difference between the approaches. If we set omitempty then if a users sets `backoffBaseSeconds: 0` there will be no value under `backoffBaseSeconds`. Hence I'll rather stick with the previous approach. It doesn't enforce defining these values on user, but assures us that there will always be value under this field.

Reply:
> If we set omitempty then if a users sets backoffBaseSeconds: 0 there will be no value under backoffBaseSeconds.

IIUC, in that case, the 0 will be represented in the field since this field is a pointer, and it can handle 0 as a separate one from the default integer value.

Reply:
Fixed